### ISSUE TITLE
netinstall -- how do we want to organize it?

### ISSUE THREAD

**Issue by udeved** [manjaro/release-plan#84]
I was wondering, now all the stuff is in place, manjaro-tools-0.13.4+ support it, CAL modules work.

**How do we want to approach netinstalls?**

We have so many options.

The basic working is, a [my_packages].yaml file is uploaded somewhere, and CAL will get the package groups from the url.

From one iso fits all DE, to profile based yaml url, everything is possible.

I would encourage to play around with 

```
netinstall=true
```

first in profile.conf.

I have created a handy script, that can transform a profile DE package list into a packages.yaml.
It can be used to update the yaml files, after you changed the Packages-[DE] file

The script is called **check-yaml**.
It can also generate CAL configuration files, and you get a complete set of yaml files that would be used on iso, or uploaded for netinstall. What the script does, is to generate the same files as buildiso does, but without building the iso.

Ideas here please.

**Comment by oberon2007** [comment]
Thanks a lot! exciting!! I will start playing very soon... :smile: Just first I will polish my profiles and see that everything is in place and working in the live-environment. Then I will check the new possibilites with pleasure.

**Comment by udeved** [comment]
Polishing live session is secondary, and it may take a while until calamares is en par with live config.
You can test both, reason is, once we established a clear plan, it needs code changes in buildiso.
Current implementation works, but its only meant to play around and to get familiar for team. ;)

**Comment by Ste74** [comment]
I want do a minimal de iso and do the other file via cal . I suggest the file for media or office or other but if user not want ... lollypop but prefer vlc simply uncheck or not check lollypop .. i see yesterday the check-yaml script .. no try to apply only read the help ..  and apply 
`check-yaml -p gnome-minimal -c ` but not work .. i missed something :blush:

**Comment by udeved** [comment]
```
$ check-yaml -p gnome-minimal -c
 --> Loading Packages: [Packages-Root] ...
 --> Loading Packages: [Packages-Gnome-minimal] ...
  -> Configuring Calamares ...

```

Look in your cache_dir, there is a new folder called netinstall. :)

**Comment by Ste74** [comment]
I have a dedicated partition for manjaro-tools : /build .. so i see /build/manjaro-tools/netinstall/gnome-minimal/x86_64/etcc.. 
here i see the .yaml file but  i have error ..
https://gist.github.com/Ste74/bd1ea46a27cdbb5d89ae6dbbc96bc25f

**Comment by Strit** [comment]
[qoute]here i see the .yaml file but i have error ..[/quote]

Do you have yaml installed as a package on that PC?

**Comment by Ste74** [comment]
Yes , but i use the git master branch not the regular 0.13.4 release

**Comment by udeved** [comment]
Try to install **python-yaml**

**Comment by Ste74** [comment]
:+1: works like a charme :smile:

**Comment by Ste74** [comment]
well , the list is the pkgs in old gnome-minimal pkg file right ?

**Comment by udeved** [comment]
$cache_dir/netinstall

depends what you set for cache_dir

But guys, please lets get back to actually discuss how we approcah it.
If there is trouble with scripts, open bug report .

**Comment by Ste74** [comment]
Ok , try to do a build to see the result ..

**Comment by Ste74** [comment]
@udeved open https://bugs.manjaro.org/index.php?do=details&task_id=17

**Comment by Ste74** [comment]
Now i think understand how work... first thought : consider to create a meta pkgs for multimedia , office and other but not equal for all .. example i3-multimedia-manjaro ecc .. and second the full and hybrid contain the same pkgs the difference is minimal+all downloaded file = full but at this point the full is necessary ?

**Comment by philmmanjaro** [comment]
Remember to use **calamares-dev** for this. The regular **calamares** package don't ship yet all needed modules for netinstallation ...

**Comment by Ste74** [comment]
@philmmanjaro , yes , i read your note in our dev-list .. i like play with this

**Comment by oberon2007** [comment]
I think it would be useful to have a way of controlling which combinations of selections will be possible and/or obligatory. For example it should not be possible to attempt an install without group ``base`` selected. Also it would make sense to exclude conflicting groups whenever this might be needed.

**Comment by oberon2007** [comment]
For hybrid installs, when we offer a basic system plus additional package groups, it would be great to have sub-categories. For example an entry "Internet" with sub-choices of Firefox/Palemoon/Chromium/...

**Comment by Ste74** [comment]
Yep , this is usefull :smile:

**Comment by Strit** [comment]
Is there a dev iso around that uses this approach, or how can I test it out? I wanna see the new shiny stuffs!

**Comment by oberon2007** [comment]
@Strit https://forum.manjaro.org/t/manjaro-i3-hybrid-for-testing/9391/2

**Comment by Strit** [comment]
Nice. Not really a tiling guy, but I guess I could use that, just to test the installer. :smile:

**Comment by udeved** [comment]
@oberon2007 

~~~
/var/build/manjaro-tools/buildiso/lxqt-minimal/x86_64/lxqt-minimal-image/etc/skel/.gtkrc-2.0 exists in both 'manjaro-base-skel' and 'manjaro-openbox-settings'
~~~
Please fix this.
I think you updated base-skel.

**Comment by oberon2007** [comment]
Obviously I missed updating lxqt and ob settings aswell, sorry. Forgot that they have separete PKGBUILDs from the other profiles, sorry.
I had renamed all **gtkrc-2.0** files in the profiles to **gtkrc-2.0** and included them in the shared **gtkrc-2.0**, similar to the procedure with the other dotfiles...
I see you've rebuilt your profiles meanwhile, so should work now.

**Comment by oberon2007** [comment]
For hybrid ISOs it would be great if we could use something like:
```
- category: "Browser"
    - name: "Firefox"
     description: "Mozilla Firefox"
     packages:
          - firefox

    - name: "Chromium"
     description: "Chromium"
     packages:
          - chromium

    - name: "Midori"
     description: "lightweight browser Midori"
     packages:
          - midori

    - name: "Palemoon"
     description: "Pale Moon Browser"
     packages:
          - palemoon-bin
```

**Comment by Ste74** [comment]
Yes,my idea: main category with multiple choice is nice

**Comment by udeved** [comment]
I don't think this is happening soon.
Remember, calamares is a distro independent installer, what you see is supposed to work on many distros. We have to live for the moment what is available, what you suggest is not implemented quickly.

**Comment by philmmanjaro** [comment]
I adopted now **mhwdcfg** into our **stable** build of **CAL**. Are the other **modules** now also ready to be reviewed and merged soon?

**Comment by udeved** [comment]
@philmmanjaro 

chrootcfg and servicescfg need slight cleanup, lets just wait until I finished the yaml schemas, which shoul be done today.

**Comment by philmmanjaro** [comment]
@udeved: no pressure, simply trigger me when ready.

**Comment by udeved** [comment]
0.13.7 is now in unstable, a last 0.13.8 should follow soon, perhaps next week end.
We now have a new package, manjaro-tools-yaml.

**Comment by udeved** [comment]
Just top let you know guys, next will version will not support yaml files other than in predefined link location.
This is until we found a working solution, no extra sausages for now.

**Comment by Ste74** [comment]
At the moment i want search why budgie profile do error in efi part  just to build it. So i wait for use new calamares modules :p I use the git version and i see more new file :)

Il 21/set/2016 13:57, udeved <notifications@github.com> ha scritto:Just top let you know guys, next will version will not support yaml files other than in predefined link location.
This is until we found a working solution, no extra sausages for now.

**Comment by udeved** [comment]
@Ste74 

I have also looked for the budgie error, I have no idea what causes it, I can't find any big differences to other profiles. My theory is, its caused by a package that the budgie profile pulls.

**Comment by Ste74** [comment]
Suspect plymouth theme , is the only very difference with gnome .. i have to see

**Comment by oberon2007** [comment]
Can I ask why we would like to disable yaml files per profile? "Working solution" for what exactly you mean?

**Comment by udeved** [comment]
I don't see the need to clutter yaml files over the profiles, we can achieve the same with only one shared file. Its counter productive to allow it until we found a structure with little maintenance.

**Comment by oberon2007** [comment]
Btw, you seem to have blocked me on github. Like that I see your posts only accidentally, if at all - Not sure if this is an ideal situation for a team on a project. You decide ...

**Comment by philmmanjaro** [comment]
Seems we have some personal issues currently with some team members. I'm also against wild growing. We should work on one **yaml** file first. Sure, people can test this technic for their own creations, however we have first to stabilize it. If I'm correct the work should go into the file [netinstall.yaml](https://github.com/manjaro/manjaro-tools-iso-profiles/blob/master/shared/netinstall/netinstall.yaml). There I see following groups for now:

- base
- base-devel
- plymouth
- xfce
- plasma
- deepin
- gnome
- i3
- lxde-gtk3
- lxqt
- mate-gtk3

For my taste already to much profiles. I'd like to ditch **deepin** (not stable yet to be widely used), **i3** (not a desktop, it's a WM), **lxde-gtk3** (who is using that). We have to start with some stable profiles which might work as given.

Also we need a proto type to test it with some community members. **ALL** the shipped profiles on the **netinstall** disk should work and that flawlessly. Only then we can adopt other profiles to the group. And if the **base** group conflicts in some kind we have to check how we deal with that. Maybe some packages from base should then be moved to the profiles needed those packages.

Also the count of each group is to high with packages. I recommend to use package groups here or if needed meta packages.

Currently everybody is working on their profiles to be the first able to release some, however the official **netinstall** media never gets released due that fact. Rather than that we are arguing internally why we decided to block custom **yaml** files. First we have to get the basics right and create a standard we can build on.

That out, I'll use now the sources not modified and as given and create some new sets of install medias. These we have to test and collect the feedback. Then we fix the conflicts and issues together. Otherwise we never finish our task at all. Hope all will agree on this approach.

**Comment by oberon2007** [comment]
This makes sense for the **netinstall**. How about **hybrid**? They seem to only make sense per profile, since they use the squashed image of the live environment and just add pkg-groups for the target install for the given environment only... I'd love to start working on **xfce-hybrid** if you agree.

**Comment by udeved** [comment]
My thought were following when doing the prototype yaml files.

hybrid yaml is for the case, where we extract sqfs images and additionally pull in packages, its ideal for shared packages accross profiles, eg java.

netinstall yaml is for full netbootstrap with chrootcfg, ie the target root is empty of packages.
The ideal candidate to host a netinstall would be either lumina or lxqt @philmmanjaro 

Generally, my idea would be to cut down on the Packages lists, only let them have basically DE pacman group packages, and pull the extra from hybrid yaml, This way, we would get identical systems with hybrid or netinstall.

So a big task in my view is to clean the lists, and we should not forget than an installer is not a fully fledged package manager, its supposed to install a stable basic system, with or without graphical session doesn't matter. The Target system should only have packages installed that are truly required and needed.

**Comment by philmmanjaro** [comment]
Exactly. Antergos needs to re-release their installer disk whenever Arch is changing some packages. So it is a constant fight against upstream. Since we can decide on our own and being independent, we have the luck to organize it for the better.

Don't forget the endusers here. They expect it fully working. So I recommend to concentrate on the bigger task first and do the official NetInstall disc first. For me either LxQT or Lumina is fine. What ever is best to display CAL in fullscreen. Maybe just a WM would be also enough to get a small disc. Was **eggwm** a possibility for that approach?

When that done and out we can go for the hybrid stuff. Hybrid might be harder to do, as you have an old base and newer packages in the end. I'd rather want to ditch that approach completely as I see some caveats already without even starting it. Sure, maybe possible but in a rolling release model it might create unwanted issues and has to be proven to work still against the odds.

Anyway, we should start with it, else the whole time we invested so far was wasted after all.

**Comment by udeved** [comment]
@philmmanjaro 

Perhaps it went a bit unnoticed, but look here...

```bash
$ check-yaml -h
Usage: check-yaml [options]
    -p <profile>       Buildset or profile [default: default]
    -a <arch>          Arch [default: auto]
    -k <name>          Kernel to use[default: linux44]
    -i <name>          Init system to use [default: systemd]
    -c                 Check also calamares yaml files generated for the profile
    -g                 Enable pacman group accepted for -p
    -v                 Validate by schema
    -q                 Query settings
    -h                 This help
```

We can already sync between lists and write yaml from profile. Its easy to expand, eg merging generated lists whatever. :)
Thought is, who the hell is supposed to maintain Packages lists and yaml files and keep them in sync? We need some tool there, or we perhaps switch to pure yaml based lists later, and do package loading in python. Bash is not so good with yaml.

**Comment by philmmanjaro** [comment]
I agree that automatisation is key here. If that approach works, why not. However I'd recommend to release some to the public so we can get their prospective also. Based on their feedback we can go on. I think the basis we have already created. It only needs to be verified. Are the current CAL modules ready for the current approach or is there still some optimization needed?

**Comment by udeved** [comment]
The packages cal module, used by hybrid is not ready for showtime in my view.
No progress update when bigger data transfer is done is problematic for end user, he may conclude the installer has crashed or something.
We would also have to add an update call so outdated preinstalled packages wopuld get updated too when pulling hybrid groups.

**Comment by philmmanjaro** [comment]
Our stable CAL misses **chroot** module, or is it feature complete for the normal **netinstall** profile?

**Comment by udeved** [comment]
I guess its working fine, and atm, I have nothing to add feature wise.
We may need to tweak here and there for some corner case error handling, and the module is naturally arch/manjaro specific, it only supports pacman.
We could add support for other PMs later, granted that PM has also some capability to change the target root like pacman. 
I know portage can do that, but it would not be a viable option to install a gentoo this way, I would rather write a module that does a stage 3 tarball extract.

**Comment by udeved** [comment]
Servicescfg is also fine, staus is, openrc install is en par with its systemd counterpart.

Just didn't have time yet to rebuild cal dev for testing, but last commit shouldn't have done any harm.

**Comment by philmmanjaro** [comment]
The latest **CAL-DEV** includes both new modules already. I simply need the go to merge it also to the stable release we have. Can be done with **v2.4.1.2** or so ...

**Comment by udeved** [comment]
No hurry, dev package is fine for now I think.
I updated pacman-mirrors package and livecd with @Kirek nice work on geoip ranking. Works fine in live session so far, and I always get fastest mirror, but it needs some further testing I haven't done yet for different language selection.

**Comment by philmmanjaro** [comment]
Which profile is it anyway we are currently working on to replace the **net** profile with the **netinstall** approach ...

**Comment by udeved** [comment]
Basically you can use any profile for netinstall, but I played with lumina and lxqt-minimal.

Set unpackfs=false and netinstall=true, that's all it needs, the rest is just how much live session you want to build as bloat. :)

My idea was, to use as slim as possible iso setup for netinstall, which is probably lumina.
Btw, I changed lumina-desktop to andromeda-icons, unfortunatley, andromeda misses and icons and fallback is oxygen. I read yesterday deeper in lumina documentation.

**Comment by philmmanjaro** [comment]
Ok, seems that is why we had that depend at first. Anyway, **lxqt** got now also released. Lets see if **v0.11** is also good to be used. The smaller the better. However we only need something to display CAL in full screen ...

**Comment by udeved** [comment]
Regarding lumina icons, they gonna have their own icon set, any icon theme serves the purpose according to docs. Easiest fix was to add the few missing icons to andromeda, because the package is considerable smaller than oxygen.

**Comment by Ste74** [comment]
Uhmm .. i read all comment .. at the moment i build only complete iso .. i'm in standby with the net or hybrid iso .. if the priority is check the net install profile for me is ok .. i can adjust group for gnome and budgie . I like start when the guideline is decided not before .. this is the reason because i'm quiet :smile:

**Comment by udeved** [comment]
https://github.com/manjaro/manjaro-tools-iso-profiles/blob/master/shared/netinstall/netinstall-systemd.yaml

This should do for a good start? Every group before base-devel is hidden and selected by default.

**Comment by udeved** [comment]
If we settle with 2 qt and 2 gtk DEs initilally, we should work in parallel to resolve the settings package conflicts for these DEs.

**Comment by Ste74** [comment]
I can set the big foot ( budgie is equal) the father of gtk :p :D
Il 24/set/2016 20:14, udeved <notifications@github.com> ha scritto:If we settle with 2 qt and 2 gtk DEs initilally, we should work in parallel to resolve the settings package conflicts for these DEs.

**Comment by philmmanjaro** [comment]
[netinstall-systemd](https://github.com/manjaro/manjaro-tools-iso-profiles/blob/master/shared/netinstall/netinstall-systemd.yaml) seems to be a good start. When can we expect a **prototype** out of it? I think **x86_64** version is enough for the beginning. We should test it then thru and thru and work with the results. All given profiles on the disk should be installable. Since **lxqt 0.11** just came out, we have to double check what might have changed.

After that we can go on and see if for example **deepin** or **i3** can be added to that concept. Maybe **budgie** and **lumina** would be also a good option. But first things first:

- prototype out for testing
- collecting the feedback from internal and public testing
- discussion with the team what our next move would be
- further development

When we have a solid base we should decide together on how we deal with the given issues we might find out by this approach ...

**Comment by udeved** [comment]
I added new netinstall profile.

https://github.com/manjaro/manjaro-tools-iso-profiles/tree/master/official/netinstall

The first testing yesterday evening was disappointing in terms of size, ~600MB

**Comment by philmmanjaro** [comment]
Ok, what are the PROs and CONs with it?

**Comment by udeved** [comment]
Pro, we just need one, only one netinstall media out. It boots very fast, no mhwd repo, only root image and live image, We could even extract the root image and cover hybrid with it.

Con, size is still too big, needs to shrink.

**Comment by philmmanjaro** [comment]
Ok, let me build it and we see what I might find.

**Comment by udeved** [comment]
I am currently not sure the fluxbox works, the skel is directly in live-overlay.
We might switch for eggwm, but I didn't manage to configure eggwm yesterday to start a maximized CAl window.

**Comment by philmmanjaro** [comment]
@udeved: maybe we have to add a feature to CAL to have that as option. Does **eggwm** work in any kind?

**Comment by udeved** [comment]
I tried eggwm in lxqt and its nice.
But I haven't found any configuration options except for the theme.

**Comment by philmmanjaro** [comment]
It doesn't have any other than that. Does it has the option for moving the application and maximize it by double click or was the window boarders missing?

**Comment by udeved** [comment]
It has afaik double click in lxqt.
But what I mean, my attempts to start eggwm as session failed, even with a created xsession desktop file.

**Comment by philmmanjaro** [comment]
Hmm, possible. What about **kwin**? Most of the framework stuff gets pulled by CAL anyway ...

**Comment by udeved** [comment]
That was a possibility I considered, but it got too late yesterday. We could try one, since there is no size gain when using fiddling stuff, since CAL depends on frameworks anyway.

**Comment by philmmanjaro** [comment]
Ok, lets give it a try. I'll test the current profile now. Keep me updated on the process. Thx.

**Comment by udeved** [comment]
delete the .fluxbox folder in skel before build, something is screwed up there and fluxbox crashes.

**Comment by philmmanjaro** [comment]
Hmm, maybe @oberon2007 can help here... Let me build it as is and we will see.

**Comment by oberon2007** [comment]
It's kind of amusing that we appear to be suddenly interested in frickin' window managers :laughing: 
**fluxbox** profile and desktop-settings should still be functional as they are in the profile and settings package. I can't check the current prototype and its fluxbox issues but I can do that during the week with pleasure.

**Comment by Ste74** [comment]
@udeved , i build a lxqt minimal for test but i have to modify the profile ? Calamares don't find the .yaml for download online set ... and post here or in other location ?

**Comment by philmmanjaro** [comment]
Let us first concentrate on the **netinstall** profile before we fiddle with something else. I'm almost finished to get a first **prototype** online ...

**Comment by Ste74** [comment]
:smile: i know but this request is because yesterday @udeved ask me to test lxqt-minimal profile with calamares for downloading gnome set and debug it :wink:

**Comment by philmmanjaro** [comment]
@Ste74: please use the **netinstall** profile now. I managed to get it kinda working. However the wallpaper gets not drawn, which doesn't matter for now. CAL starts in fullscreen ...

**Comment by Ste74** [comment]
Ok thank you no problem build the netinstall and try it :wink:  ..

**Comment by philmmanjaro** [comment]
@Ste74: seems some is off with **netgroups**. The default can't find it either. Have to check now the configs.

**Comment by udeved** [comment]
Make sure you pull profiles and rebuild tools.
I had to change url of netgroups to make it work with a netinstall profile.

**Comment by philmmanjaro** [comment]
I adjusted the profiles once more. Seems to be very experimental. It depends most likely on the mirror which got chosen. So this is key to debug first. On first run I was not able to install a working Manjaro ...

**Comment by philmmanjaro** [comment]
I might have now a working **netinstall** disk for internal testing. However following issues I've seen:

* seems the African mirror gets chosen, so we have to do some mirror-ranking to get a faster server
* using the fullscreen option of **fluxbox** we have to ALT-TAB thru the windows if more than one open of CAL
* when **chrootcfg** fails at some point the whole chain fails
* the network groups need to be constantly checked and tested, otherwise a new **netinstall** media is needed

In short, it is still a long way to go. Maybe we should ask @teo to add a fullscreen mode to CAL.

**Comment by philmmanjaro** [comment]
Currently it fails with `Sudoers file not writeable`. Missing package?

**Comment by philmmanjaro** [comment]
With https://github.com/manjaro/manjaro-tools-iso-profiles/commit/061642adf8718dbb320c9471bbd86de377dc58af I managed to get rid of the bug. Now I was able to successful install **xfce** on the target. However **lighdm** was not configured as **sddm** was selected. We should find a way that **chrootcfg** readjust the displaymanager values based on the selection. @udeved: any idea how we can achieve that in CAL?

**Comment by philmmanjaro** [comment]
Hmm, seems I found a way. Currently we only offer **xfce**, **plasma5**, **lxqt** and **gnome**. Therefore I adjusted the **displaymanager.conf** file like this on my personal build:

```
---
displaymanagers:
  - gdm
  - lightdm
  - sddm

#defaultDesktopEnvironment:
#    executable: "startfluxbox"
#    desktopFile: "fluxbox"

basicSetup: false
```
Maybe we should find a more dynamic way for it ...

**Comment by philmmanjaro** [comment]
After customizing **services.conf** it seems to work now. At least **xfce** I've tested now.

```
services:
    - name: bluetooth
      mandatory: false

    - name: cronie
      mandatory: false

    - name: ModemManager
      mandatory: false

    - name: NetworkManager
      mandatory: false

    - name: tlp
      mandatory: false

    - name: tlp-sleep
      mandatory: false

    - name: gdm
      mandatory: false

    - name: lightdm
      mandatory: false

    - name: sddm
      mandatory: false

targets:
    - name: "graphical"
      mandatory: true

disable:
    - name: pacman-init
      mandatory: false
```
The current ISO can be [downloaded here](https://sourceforge.net/projects/manjarotest/files/16.10-dev/netinstall/)...

**Comment by udeved** [comment]
I think we would need to make displaymanager in yaml optional, and implement a chroot detection.

I know the progreebar in chrootcfg is far from ideal. Problem is, afaik, you only have 5% of the entire progress bar available, and this is scaler from 0.0 to 1.0, ie the 5% equal 100% in the module. @philmmanjaro 
Whatever I tried, the progressbar stops growing once it reached value 1.0.

**Comment by philmmanjaro** [comment]
Hmm, maybe we should ask @teo about it. For now I recommend to concentrate on the given issues we have with the more dynamic approach. I hope **lxqt** and **plasma** is equal working as **xfce**. Maybe someone can give my ISO a try in that regard. We should work together on the profile. Maybe using **kwin** might be not so bad after all. Also if possible shrinking of the image ...

**Comment by udeved** [comment]
Yes, we should concentrate on the one netinstall profile. Whoever finds the best configuration, we go with it. I think the general approach is working, ie with no mhwd repo and DE layer.

**Comment by udeved** [comment]
@philmmanjaro 

Did you encounter a case where chrootcfg failed?

**Comment by philmmanjaro** [comment]
@udeved: for now it worked. Depending on the mirror and which packages got selected. I only had to adjust some to get it working. Seems to be fine now.

**Comment by udeved** [comment]
Did you have mirrors-live service in live array? And did you build unstable?
Its strange, because kirek's geoip changes made it in the mirrors-live service, and my test builds all had German mirrors ranked initially after boot.

**Comment by udeved** [comment]
Its funny though, that a lxqt-minimal is only slightly bigger than our current netinstall setup.
We could also consider lxqt as base and eggwm alternatively to kwin.
lumina is nice but I think its need more to to mature and implement the stuff in the lumina docs, mainly wm and powermgmnt

**Comment by philmmanjaro** [comment]
I used the profile as given. **mirror-live** is there. I managed to test additionally to **xfce** also **plymouth** and **plasma**. So far so good. I had a default basic KDE setup. Will now test **lxqt** and **gnome** if I find time for it...

**Comment by philmmanjaro** [comment]
However the service seems to fail if not the network connection is established. We might need to make sure of that first ...

**Comment by udeved** [comment]
I think the service is probably best changed in the unit depend. It works fine on openrc at boot.

**Comment by Ste74** [comment]
I can update the profile for net  and test gnome installation or had to know something ?

**Comment by udeved** [comment]
You can update your profile or build netinstall profile. The netinstall yaml are read from url @Ste, they are not like other conf file on iso. ;)

**Comment by Ste74** [comment]
ok so can build a netinstall profile and try to see how gnome install

**Comment by philmmanjaro** [comment]
@Ste74: I just tested [my ISO](https://sourceforge.net/projects/manjarotest/files/16.10-dev/netinstall/). Gnome and all other profiles are good. Simply give it a spin.

@udeved: XFCE and Plasma5 are visualize Ok as the the stock config is fine. LxQt misses some icons and Gnome has the usual issues in VirtualBox with **gdm**. All installations went as expected. Now we have to polish the **yaml**, check if we get a better and smaller live-image which doesn't look so like '95 and we are good to go. We have to find a better way to get the mirrorsrank established. Seems **systemd** has there some issues.

Other than that it seems we are on the right track. On a later step we might include also the Manjaro theming for these profiles. However for now it is good as it is now.

**Comment by udeved** [comment]
Perhaps is also good move to get Packages lists of the profiles in line with heir yaml counterparts.
check-yaml writes from list to yaml, not yaml to list.

**Comment by Ste74** [comment]
Well downloaded and run..  waiting install gnome in wm @philmmanjaro  spin and note that on calamares : if another language is selected calamares set always system language in english and also the keyboard map is not set automatically on the new lang..

**Comment by udeved** [comment]
@philmmanjaro 

This should enable DM detection for netinstall
https://github.com/manjaro/manjaro-tools/commit/3784265d8c5aef7ee22e7362a46612a9748f2847

For the DM service, we just need to write the DM from globalstorage and don't write DM in services.conf

**Comment by udeved** [comment]
@philmmanjaro 

I think the displaymanager thing is probably best addressed in a new module rather than tweaking existing DM module. Its pretty arch/manjaro specific what we have to add, especially if we want to support plymouth DM services. I am not sure how other distros solve plymouth, but we don't even have a unified approach on arch, eg sddm and lxdm don't have extra plymouth unit.
The existing DM module has DM detection, but it relies on defined DMs in job config before it checks and writes them in global storage.

So how do we deal with this? Rewrite existing module or add a new module?

**Comment by oberon2007** [comment]
I've played a little with the cosmetical manjarisation of the current fluxbox prototype:
![](http://imgur.com/ufYLX7l.png)
![](http://imgur.com/HxHlPf3.png)
![](http://imgur.com/pzWPayC.png)

**Comment by oberon2007** [comment]
Now including **cantarell-fonts** and **xcursor-breeze** the ISO is 603MB in size. I doubt that we can get much lower than that.
**kwin** seems to be not an alternative, either. Adding the package alone already results in an ISO of 630MB. So I didn't dig deepin into configuring with that.
In any case we already have an environment free of **gtk2**. **gtk3** is only needed as a depends of nm-applet but I guess there is not really a way around it?
I've noticed that vmware support is quite heavy because via **opencl-mesa** it pulls in **gcc** which alone is

**Comment by oberon2007** [comment]
I've just tried replacing **networkmanager** with **connman** and **cmst**. That makes the ISO quite a bit smaller with no gtk2 or gtk3 needed. It's pure **qt** now and only **566MB** in size. Shall we try that?

**Comment by oberon2007** [comment]
Also, go we really need a DM at all? Why not just autologin with startx?

**Comment by Strit** [comment]
Maybe it's possible to just start Calamares from CLI opening an X window in full screen?
Like: https://bbs.archlinux.org/viewtopic.php?id=107319

**Comment by oberon2007** [comment]
**fluxbox** all in all is about 3MB, maybe not worth the bother to get rid of the WM.... ;)

**Comment by oberon2007** [comment]
Just made another test with **jwm** instead of **fluxbox** and without a DM. All that just saves a few MB, so I guess Maybe we can just stay with the current configured version... Using **connman** would make sense though as explained above. We should just make sure it is reliable enough.

**Comment by udeved** [comment]
NM is a depend of calamares networkcfg.

**Comment by philmmanjaro** [comment]
Right. We need to write a new module if we want to use something else as **NM**.

**Comment by udeved** [comment]
I thought too complicated and missed the obvious to address netinstall DM issue.
https://github.com/manjaro/manjaro-tools/commit/7d4bb08b0693cfd0371b068c0bbaa169b0df6e5a

**Comment by udeved** [comment]
Ok, now the services CAl module, I'll add something to support it based on autodetection of DM module.
However, we can support plymouth, but we can only start standard service of DM. 
eg lightdm-plymouth.service vs sddm.service

**Comment by excalibur1234** [comment]
i remember when @Esclapion and i tested connman with cmst as an alternative to networkmanager over a year ago. we switched back to networkmanager, because cmst (which is the best GUI for connman) was not able to connect to all types of wifi networks. i do not know, whether this has changed since then. i know that cmst is still being actively developed, though.

we can just test with connman and cmst until people start to complain...

**Comment by udeved** [comment]
connman won't work with calamares, its not only the networkcfg module, but calamares network online  detection depends on networkmanager.

**Comment by oberon2007** [comment]
I've made the fluxbox-theme more **vertex-maia**-isch now:
![flux_maia](https://cloud.githubusercontent.com/assets/11800714/18956945/9071cf26-865d-11e6-87d0-fcc4a4c6f33e.jpg)

**Comment by philmmanjaro** [comment]
Was the image updated and retested? If not, please use rsync and update it in **manjarotest** project.

**Comment by oberon2007** [comment]
Can do that later tonight.

**Comment by oberon2007** [comment]
Updated the image and included build log. Used https://github.com/manjaro/manjaro-tools/commit/8bdd63d522cbdab5b0e340e87329604396bd986d.
I successfully installed an lxqt with plymouth in vbox with this image.

**Comment by oberon2007** [comment]
I think we should remove the fullscreen setting for calamares and switch back on window decorations. Also I'm wondering if it would be best to have calamares launched from the terminal by default so during installation people will see some progress during package download
![](http://i.imgur.com/fy1fpOu.png)

**Comment by udeved** [comment]
The boot menu restructuring is in progress.
I swapped isolinux.cfg for syslinux modular config.
This should improve creating live usb in bios mode.
No changes to efi loader yet, I am not sure syslinux can boot iso9660 (in efi or bios)
[![syslinux.png](https://s14.postimg.org/rwwf37bip/syslinux.png)](https://postimg.org/image/6ahem6cy5/)

Now the menu needs some color tweaking, and position tweaking, and I haven't found out yet how to display resolution and language menu in syslinux.

**Comment by philmmanjaro** [comment]
Modular Configs don't support the same features as Suse had designed it with **gfxboot**. **Modular** is the standard way of **syslinux** to create a menu. **gfxboot** on the other hand is more advanced and has the possibility to use **languages**. However I don't know if it works with **efi**. We might need to check with a newer version of **gfxboot**.

**Comment by udeved** [comment]
My attempts to use the gfxboot with syslinux(bios) failed so far.
Its supposed to be working.
We could rid of the isolinux packages, and have something like syslinux-overlay if necessary, syslinux seems to have absorbed all isolinux features afaik.
modular config seem nice, but I only have limited experience with syslinux.

**Comment by philmmanjaro** [comment]
I assume with unified bios/efi menu setup ...

**Comment by udeved** [comment]
Yes, plan is is default to syslinux, the won't be any isolinux dir in the cd root.
I just need to find a proper configuration.
After this is in place, it should be easy to experiment with efi syslinux again.
The syslinux wiki says, we can put in isolinux.cfg any time, and it will be preferred, eg for gfxboot eventually.

**Comment by udeved** [comment]
I restored the syslinux theme back to normal, but with modular cfg files and gfxboot.

**Comment by philmmanjaro** [comment]
Also a great feature to have.

**Comment by udeved** [comment]
@philmmanjaro 

Even better would be to have the gfxboot tool from suse ported. Do you have time to create a PKGBUILD?

I am currently trying to find out what en.tlk file is supposed to do, since some file is bootlogo cpio archive is responsible for text display of the F-keys

**Comment by philmmanjaro** [comment]
?!?, the theme is [here](https://github.com/manjaro/artwork-gfxboot). and you can install [gfxboot](https://github.com/manjaro/packages-extra/tree/master/gfxboot) from our repositories. Currently we ship **v4.5.0**.

**Comment by udeved** [comment]
Clic on the link, suse has a little cli tool to set and test gfxboot.cfg

**Comment by philmmanjaro** [comment]
Do you mean the shell script **gfxtest**? I don't have included it yet to the current **gfxboot-4.5.15-1** build. However you can also test it directly with **gfxboot** if you find a way how to do so ...

**Comment by udeved** [comment]
Anyway, would be nice if someone could fine tune the [syslinux-gfxboot-theme](https://github.com/manjaro/packages-extra/blob/master/syslinux-themes/PKGBUILD).
Its ugly atm, and I am not good with this.

**Comment by philmmanjaro** [comment]
What do you mean with ugly. Do you have a screenshot of it?

**Comment by udeved** [comment]
![syslinux1](https://s13.postimg.org/p47j03ak7/syslinux1.png)
![syslinux1](https://s13.postimg.org/ndohylt13/syslinux2.png)

**Comment by Ste74** [comment]
@udeved you like maia color instead blue ? and the selection bar with how color ?

**Comment by udeved** [comment]
I pretty much don't care for the look, I meant with ugly, that I copied binary stuff from suse instead of compiling. Anyway, feel free to take on colors and stuff, not my cup of tea. ;)

**Comment by udeved** [comment]
So, what are we gonna do in terms of netgroups yaml?
With advancement of CAL packages module, can we maintain centralized file? @manjaro/core-tools @manjaro/maintainers

**Comment by oberon2007** [comment]
Sounds good! CAL just still needs to find a solution for a proper [progress bar](https://github.com/calamares/calamares/pull/281#issuecomment-277716594) before out becomes tacky usable. What's your idea of a structure in regards of package distribution in yaml files?

**Comment by udeved** [comment]
I'd keep it at minimum, if possible centralized for all profiles, as opposed to yaml per profile.
We eventually split the yaml in qt and gtk for example, but we can also keep them together.

**Comment by udeved** [comment]
For example, chrootcfg may be better with profile based yaml in the end, but its good for testing to have it support many DE. 
My idea is to have lxqt chrootcfg out, which lasts a while. CAL would only offer to instal lxqt in this case.

**Comment by udeved** [comment]
Will be profile based netgroups, added with mt-0.14.4

**Comment by Ste74** [comment]
@udeved i used check-yaml 

> 
> /build/manjaro-tools/netinstall/gnome/x86_64 >>> ls                                                     
> Packages-Desktop-x86_64-systemd.yaml  Packages-Root-x86_64-systemd.yaml
> /build/manjaro-tools/netinstall/gnome/x86_64 >>>  

i have to modify like yours in lxqt ?

**Comment by udeved** [comment]
Yes @Ste74 modify as you like.
It just gives you profile packages.

**Comment by Ste74** [comment]
also the name of file ?

**Comment by udeved** [comment]
yes, name must match repo name.

**Comment by Ste74** [comment]
Oh Oh :p  :(  my access is not granted to calamares-netgroups repo .. @philmmanjaro , @Kirek can you do please ?

**Comment by philmmanjaro** [comment]
@Ste74: rights are adjusted.
